### DISCUSSION TITLE
配置自定义路由无效，官方文档说可以，实际不行，求助

### DISCUSSION THREAD

**Discussion by lesterlong007** [NervJS/taro#15195] [Q & A]
### 相关平台
H5

**浏览器版本: all**
**使用框架: React**

### 复现步骤
在app.config.js 配置路由
现有配置如下：
import { pages, subpackages } from './root';

export default defineAppConfig({
  pages,
  subpackages,
  subPackages: subpackages,
  window: {
    backgroundTextStyle: 'light',
    navigationBarBackgroundColor: '#fff',
    navigationBarTitleText: '',
    navigationBarTextStyle: 'black',
  },
  usingComponents: {},
});

需要加个routes 来配置路由，来指定路径 和 路径对应的文件，而不是默认文件路径。
官网说：基础库 3.160.3 及以上版本支持， 但是实际根本不行

### 期望结果
可以配置自定义路由

### 实际结果
配置自定义路由无效，且ts会报错警告

### 环境信息
```
Taro v3.6.6
Taro CLI 3.6.6 environment info:
    System:
      OS: macOS 14.2.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 16.16.0 - ~/.nvm/versions/node/v16.16.0/bin/node
      Yarn: 1.22.15 - ~/.nvm/versions/node/v16.16.0/bin/yarn
      npm: 8.11.0 - ~/.nvm/versions/node/v16.16.0/bin/npm
    npmPackages:
      @tarojs/cli: 3.6.6 => 3.6.6 
      @tarojs/components: 3.6.6 => 3.6.6 
      @tarojs/helper: 3.6.6 => 3.6.6 
      @tarojs/mini-runner: 3.6.6 => 3.6.6 
      @tarojs/plugin-framework-react: 3.6.6 => 3.6.6 
      @tarojs/plugin-platform-alipay: 3.6.6 => 3.6.6 
      @tarojs/plugin-platform-h5: 3.6.6 => 3.6.6 
      @tarojs/react: 3.6.6 => 3.6.6 
      @tarojs/runtime: 3.6.6 => 3.6.6 
      @tarojs/shared: 3.6.6 => 3.6.6 
      @tarojs/taro: 3.6.6 => 3.6.6 
      @tarojs/webpack-runner: 3.6.6 => 3.6.6 
      @tarojs/webpack5-runner: 3.6.6 => 3.6.6 
      babel-preset-taro: 3.6.6 => 3.6.6 
      eslint-config-taro: 3.6.6 => 3.6.6 
      react: ^18.0.0 => 18.2.0 
```

---
### 补充信息
none

**Comment by ZakaryCode** [comment]
> 在app.config.js 配置路由 现有配置如下： import { pages, subpackages } from './root';
> 
> export default defineAppConfig({ pages, subpackages, subPackages: subpackages, window: { backgroundTextStyle: 'light', navigationBarBackgroundColor: '#fff', navigationBarTitleText: '', navigationBarTextStyle: 'black', }, usingComponents: {}, });

请提供完整示例项目，仅以此无法进一步定位实际问题。

> 需要加个routes 来配置路由，来指定路径 和 路径对应的文件，而不是默认文件路径。 官网说：基础库 3.160.3 及以上版本支持， 但是实际根本不行

基础库版本应该不是 H5 端会存在的内容，请明确表述

**Comment by lesterlong007** [comment]
压根不存在呀，defineAppConfig 的配置项，定义的ts Config interface 都没有routes属性，只是官网在骗人说有而已，实际根本不存在

**Comment by lesterlong007** [comment]
官网的app.config.js示例是这样：
export default {
  pages: ['pages/home/home', 'pages/list/list', 'pages/detail/detail'],
  subPackage: [
    {
      root: 'packageA',
      pages: ['pages/home/home', 'pages/list/list', 'pages/detail/detail'],
    },
  ],
  routes: [
    {
      // 投放入口，scheme中的path
      path: 'home',
      // 真实的物理存储路径
      page: 'pages/home/home',
    },
    {
      path: 'list',
      page: 'pages/list/list',
    },
    {
      path: 'foo/bar',
      page: 'pages/list/list',
    },
  ],
}

但是实际根本不行，连ts定义的interface也不存在routes属性，官网对应的文档链接如下：
https://docs.taro.zone/docs/app-config#routes

**Comment by ZakaryCode** [comment]
稍微看清楚文档一些，这里标注的是百度智能小程序专属属性

<img width="451" alt="image" src="https://github.com/NervJS/taro/assets/24262362/646a0874-9eba-4931-bdaf-479282a5bf8e">

h5 参考[这里](https://docs.taro.zone/docs/config-detail#h5routercustomroutes)

**Comment by lesterlong007** [comment]
sry, 没注意只针对百度小程序，h5.customRoutes 没有啥意义，只是重命名了path, 并不是像routes 这样可以自己指定路径和文件

**Comment by jiucheng-front** [comment]
解决了吗？ h5配置自定义路径也是无效
